src/UploadedFileFactoryInterface.php

```diff
@@ -18,22 +18,22 @@
      * @see http://php.net/manual/features.file-upload.post-method.php
      * @see http://php.net/manual/features.file-upload.errors.php
      *
-     * @param string|resource $file
-     * @param integer $size in bytes
-     * @param integer $error PHP file upload error
-     * @param string $clientFilename
-     * @param string $clientMediaType
+     * @param string|resource|StreamInterface $file Underlying file, PHP stream
+     *     resource, or StreamInterface representing the uploaded file content.
```

Comment:
Maybe strictly `StreamInterface`? The ambiguity of `string $filename` vs `string $filecontent` worries me. We can remove that if we ensure that streams are always used, since stream factory has separate methods for each construction.

Reply:
Strongly agree with this - not even so much because of the ambiguity, but because of feature envy. This competes with the stream-factory interface, and directly leads to inconsistencies in consumer code.

Reply:
This makes sense to me. That said: I propose that `UploadedFileInterface` extend `StreamInterface`. The rationale is that you will need a stream instance to create an `UploadedFile` instance, which is the same rationale I used for extension in the other interface cases. Thoughts?

Reply:
Makes sense to me:

```php
$stream = $uploadFactory->createStream($content);
$upload = $uploadFactory->createUploadedFile($stream, ...);
```

Reply:
Ironically, I had the php-fig PR _already_ noting that `UploadedFileFactoryInterface` extends `StreamFactoryInterface`; I just hadn't taken the extra step of changing the intial argument to `createUploadedFile()` to a stream.

Updated in both patches now.